pkg/controller/certificates/certificate_controller.go

```diff
@@ -209,12 +211,14 @@ func (cc *CertificateController) syncFunc(key string) error {
 // it's spammy and usually user error. Instead we will log these errors at a
 // higher log level. We still need to throw these errors to signal that the
 // sync should be retried.
-func IgnorableError(s string, args ...interface{}) ignorableError {
-	return ignorableError(fmt.Sprintf(s, args...))
+func IgnorableError(s string, args ...interface{}) NewIgnorableError {
+	return NewIgnorableError(fmt.Sprintf(s, args...))
 }
 
-type ignorableError string
+// NewIgnorableError error
+type NewIgnorableError string
```

Comment:
I'd prefer we don't expose types we don't want to be exposed just because golint complains. Once made public it's part of the API we'll need to support. So I'd prefer not doing this.